sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md

```diff
@@ -0,0 +1,389 @@
+
+#### `TimeoutException` 
+
+This indicates that the Event Hubs service did not respond to an operation within the expected amount of time. This may have been caused by a transient network issue or service problem. The Event Hubs service may or may not have successfully completed the request; the status is not known. It is recommended to attempt to verify the current state and retry if necessary.
+
+#### `MessageSizeExceededException` 
+
+Event data, both individual and in batches, have a maximum size allowed. This includes the data of the event, as well as any associated metadata and system overhead. The best approach for resolving this error is to reduce the number of events being sent in a batch or the size of data included in the message. Because size limits are subject to change, please refer to [Azure Event Hubs quotas and limits](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas) for specifics.
+
+#### `QuotaExceededException` 
+
+The messaging entity has reached its maximum allowable size. This exception can happen if the maximum number of consumers have already been created for a given partition and consumer group. Because  limits are subject to change, please refer to [Azure Event Hubs quotas and limits](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas) for specifics.
+
+#### `DateTimeOffset`
+
+An intrinsic .NET type, the `DateTimeOffset` represents a point in time relative to Coordinated Universal Time (UTC).  Event Hubs client library members representing date/time data are normalized to UTC and will change to using `DateTimeOffset` in the second preview to avoid the ambiguity around time zones associated with the use of `DateTime`.
+
+#### `IAsyncEnumerable<T>`
+
+An intrinsic .NET type currently in preview, the `IAsyncEnumerable<T>` enables iterating over an enumerable in an asynchronous way, allowing for an infinite sequence to be generated from a source requiring asynchronous communication.  The second preview of the Event Hubs client library makes use of this concept in its approach for allowing a "streaming consumer."
+
+### Examples: Configuring Retries
+
+#### Creating an Event Hub client with default retry configuration
+
+```csharp
+var connectionString = "<< CONNECTION STRING WITH EVENT HUB >>";
+var client = new EventHubClient(connectionString);
+```
+
+#### Creating an Event Hub client with custom retry configuration
+
+```csharp
+var clientOptions = new EventHubClientOptions();
+
+clientOptions.RetryMode = RetryMode.Fixed;
+clientOptions.MaxRetries = 5;
+clientOptions.Delay = TimeSpan.FromMilliseconds(250);
+clientOptions.MaxDelay = TimeSpan.FromSeconds(2);
+clientOptions.TryTimeout = TimeSpan.FromSeconds(90);

```

Comment:
`clientOptions.TryTimeout` is an odd name... though, I'm not sure what a better one is.